src/custom/state/claim/hooks/index.ts

```diff
@@ -1061,3 +1066,71 @@ export function useVCowData(): VCowData {
 
   return { isLoading, vested, unvested, total }
 }
+
+/**
+ * Hook used to swap vCow to Cow token
+ */
+
+interface SwapVCowCallbackParams {
+  openModal: (message: string, operationType: OperationType) => void
+  closeModal: () => void
+}
+
+export function useSwapVCowCallback({ openModal, closeModal }: SwapVCowCallbackParams) {
+  const { chainId, account } = useActiveWeb3React()
+  const vCowContract = useVCowContract()
+
+  const addTransaction = useTransactionAdder()
+  const vCowToken = chainId ? V_COW[chainId] : undefined
+
+  const swapCallback = useCallback(async () => {
+    if (!account) {
+      throw new Error('Not connected')
+    }
+    if (!chainId) {
+      throw new Error('No chainId')
+    }
+    if (!vCowContract) {
+      throw new Error('vCOW contract not present')
+    }
+    if (!vCowToken) {
+      throw new Error('vCOW token not present')
+    }
+
+    const args: { from: string; gasLimit?: BigNumber } = {
+      from: account,
+    }
+
+    const estimatedGas = await vCowContract.estimateGas.swapAll(args).catch(() => {
+      // general fallback for tokens who restrict approval amounts
+      return vCowContract.estimateGas.swapAll().catch((error) => {
+        console.log(
+          '[useSwapVCowCallback] Error estimating gas for swapAll. Using default gas limit ' +
+            APPROVE_GAS_LIMIT_DEFAULT.toString(),
+          error
+        )
+        return APPROVE_GAS_LIMIT_DEFAULT
+      })
+    })
+
+    args.gasLimit = estimatedGas
+
+    const summary = `Convert vCOW for COW`
```

Comment:
For a future PR, and with less prio, we could try to get an estimation on how much we are swapping approximately?

Reply:
i think this is absolutely not necessary. adding the amount in the modal will not be accurate and may cause MORE confusion with users saying "oh but the modal showed this but i got xx" and the fact that it's vesting linearly. i think it's better to leave it as is, with no amount

Reply:
@W3stside On a second thought, I think in practice the deviation of amounts is likely negligible for most account balances. Unless you own millions of vCOW token where it can be somewhat noticeable in mere minutes, right?